packages/bitcore-wallet-service/src/lib/server.ts

```diff
@@ -4534,6 +4534,191 @@ export class WalletService {
     return cb(null, data);
   }
 
+  moonpayGetKeys(req) {
+    if (!config.moonpay) throw new Error('Moonpay missing credentials');
+
+    let env = 'sandbox';
+    if (req.body.env && req.body.env == 'production') {
+      env = 'production';
+    }
+    delete req.body.env;
+
+    const keys: {
+      API: string,
+      WIDGET_API: string,
+      API_KEY: string,
+      SECRET_KEY: string,
+    } = {
+      API: config.moonpay[env].api,
+      WIDGET_API: config.moonpay[env].widgetApi,
+      API_KEY: config.moonpay[env].apiKey,
+      SECRET_KEY: config.moonpay[env].secretKey,
+    };
+
+    return keys;
+  }
+
+  moonpayGetQuote(req): Promise<any> {
+    return new Promise((resolve, reject) => {
+      const keys = this.moonpayGetKeys(req);
+      const API = keys.API;
+      const API_KEY = keys.API_KEY;
+
+      if (
+        !checkRequired(req.body, ['currencyAbbreviation', 'baseCurrencyAmount', 'extraFeePercentage', 'baseCurrencyCode'])
+      ) {
+        return reject(new ClientError("Moonpay's request missing arguments"));
+      }
+
+      const headers = {
+        'Content-Type': 'application/json',
+      };
+
+      let qs = [];
+      qs.push('apiKey=' + API_KEY);
+      qs.push('baseCurrencyAmount=' + req.body.baseCurrencyAmount);
+      qs.push('extraFeePercentage=' + req.body.extraFeePercentage);
+      qs.push('baseCurrencyCode=' + req.body.baseCurrencyCode);
+
+      if (req.body.paymentMethod) qs.push('paymentMethod=' + req.body.paymentMethod);
+      if (req.body.areFeesIncluded) qs.push('areFeesIncluded=' + req.body.areFeesIncluded);
+
+      const URL: string = API + `/v3/currencies/${req.body.currencyAbbreviation}/buy_quote/?${qs.join('&')}`;
+
+      this.request.get(
+        URL,
+        {
+          headers,
+          json: true
+        },
+        (err, data) => {
+          if (err) {
+            return reject(err.body ? err.body : err);
+          } else {
+            return resolve(data.body ? data.body : data);
+          }
+        }
+      );
+
+    });
+  }
+
+  moonpayGetSignedPaymentUrl(req): Promise<{urlWithSignature: string}> {
+    return new Promise((resolve, reject) => {
```

Comment:
Some of these functions like this one don't need to be a promise, right?

Reply:
Promise removed. Ty